src/components/BounceRuleDetailed/index.jsx

```diff
@@ -1,61 +1,531 @@
+    <TableBody>
+      <TableRow>
+        <TableCell>
+          <strong>ID</strong>
+        </TableCell>
+        <TableCell>{currentRule.id}</TableCell>
+      </TableRow>
+      <TableRow>
+        <TableCell>
+          <strong>Bounce Action</strong>
+        </TableCell>
+        <TableCell>{currentRule.bounce_action}</TableCell>
+      </TableRow>
+      <TableRow>
+        <TableCell>
+          <strong>Response Code</strong>
+        </TableCell>
+        <TableCell>{currentRule.response_code}</TableCell>
+      </TableRow>
+      <TableRow>
+        <TableCell>
+          <strong>Enhanced Code</strong>
+        </TableCell>
+        <TableCell>{currentRule.enhanced_code}</TableCell>
+      </TableRow>
+      <TableRow>
+        <TableCell>
+          <strong>RegEx</strong>
+        </TableCell>
+        <TableCell>{currentRule.regex}</TableCell>
+      </TableRow>
+      <TableRow>
+        <TableCell>
+          <strong>Priority</strong>
+        </TableCell>
+        <TableCell>{currentRule.priority}</TableCell>
+      </TableRow>
+      <TableRow>
+        <TableCell>
+          <strong>Description</strong>
+        </TableCell>
+        <TableCell>{currentRule.description}</TableCell>
+      </TableRow>
+    </TableBody>
+  </Table>
+);
+
+const BounceRuleDetailed = ({
+  currentRule,
+  newRule,
+  isEditClicked,
+  isChangeModalOpen,
+  isConfirmOpen,
+  isCancelConfirmOpen,
+  handleModalConfirm,
+  handleModalClose,
+  onChangeRule,
+  handleButtonClicked,
+}) => (
+  <div className="detailed-page-container">
+    <Header />
+    <Row>
+      <Column width={6} offset={2}>
+        <Breadcrumb>
+          <Link to="/bounce_rules">Bounce Rules</Link>
+          <Link to={`/bounce_rules/${currentRule.id}`}>{currentRule.id}</Link>
+        </Breadcrumb>
+      </Column>
+    </Row>
+    <Row>
+      <Column width={6} offset={2}>
+        <h1>
+          {"Bounce Rule"} {currentRule.id}
+        </h1>
+      </Column>
+      {isEditClicked ? (
+        <Column className=" sg-right" width={4} offset={8}>
+          <Button
+            onClick={() => handleButtonClicked("cancelClicked")}
+            onKeyDown={() => handleButtonClicked("cancelClicked")}
+            data-test="cancel-button"
+            className="header-button cancel-button"
+            type="secondary"
+          >
+            {"Cancel"}
+          </Button>
+          <Button
+            onClick={() => handleButtonClicked("saveClicked")}
+            onKeyDown={() => handleButtonClicked("saveClicked")}
+            data-test="save-button"
+            className="header-button save-button"
+            type="primary"
+          >
+            {"Save"}
+          </Button>
+        </Column>
+      ) : (
+        <Column
+          className="details-button-column sg-right"
+          width={1}
+          offset={11}
+        >
+          <span>
+            <Button
+              onClick={() => handleButtonClicked("editClicked")}
```

Comment:
Same here if there is a way to handle the clicks through ids/values we should refactor it